### ISSUE TITLE
Provide option to process order and go to thankyou when returning from PayPal via cart or product pages

### ISSUE THREAD

**Issue by rajeshraval786** [woocommerce/woocommerce-gateway-paypal-express-checkout#885]
Hello Team

I am suffering from the issue when I place an order from the cart using PayPal express checkout popup button and it's redirecting me back to the checkout page to confirm the order and then I press the place order button and after that my order is going to be completed earlier that issue is not there now, How can I skip place order button and redirecting them to direct on thankyou page in woocommerce.

Give me any solution, please...

Thanks in advance.

**Comment by chickenn00dle** [comment]
Hey @rajeshraval786 👋 

> I am suffering from the issue when I place an order from the cart using PayPal express checkout popup button and it's redirecting me back to the checkout page

Just to make sure we are on the same page, here is what I'm understanding the issue to be:

- The customer selects the PP button from the cart
- The customer enters payment details via the PP popup
- The customer is redirected to the order confirmation page (checkout)
- The customer confirms the order and is finally redirected to the thank you page

If this is correct, then this is actually expected behavior. The only time you should immediately be redirected to the thank you page is when placing the order from checkout. Every other time, we want to direct customers to this confirmation page to give them a chance to verify the order details and finalize the order.

**Comment by rajeshraval786** [comment]
Yes, Exactly but is there any solution for skip the confirmation page and place directly order.

**Comment by chickenn00dle** [comment]
There is not a way to change this behavior. I'm not too sure whether removing the ability for customers to confirm their order in WC is something we should be doing either.

That said, I'll go ahead and re-open this issue and make it an enhancement request. Although, as a heads up, it's unlikely we will be able to look into this in the near future.

**Comment by rajeshraval786** [comment]
So for now confirmation page is remaining as it is, we can't do anything with this plugin to skip that page and direct redirecting on thankyou page.

**Comment by chickenn00dle** [comment]
That's right.

**Comment by rajeshraval786** [comment]
But earlier we were not facing this issue on the website

**Comment by rajeshraval786** [comment]
A few Days ago, the PP button stopped working and suddenly vanished from the website. Somehow we managed to get it back. 
and after that only, this new issue occured.

**Comment by chickenn00dle** [comment]
>  and after that only, this new issue occurred.

Hmm. This shouldn't have been working this way before your PP button vanished and if it was, it may have been a bug.

When triggered from cart or product pages, we explicitly redirect back to checkout when returning from Paypal here:
https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/blob/85a478de146459418b23854cca4a60f073344dbf/assets/js/wc-gateway-ppec-smart-payment-buttons.js#L191-L198

Otherwise, when triggered from checkout, we instead manually trigger the checkout form when returning from PayPal here: https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/blob/85a478de146459418b23854cca4a60f073344dbf/assets/js/wc-gateway-ppec-smart-payment-buttons.js#L184-L189

This has been the case for at least the last year if not longer.

**Comment by rajeshraval786** [comment]
yes exactly I am getting this type of URL ?woo-paypal-return=true&token is there any solution to resolve this? please

**Comment by rajeshraval786** [comment]
and in inspect I am getting something like

<noscript>
			Since your browser does not support JavaScript, or it is disabled, please ensure you click the <em>Update Totals</em> button before placing your order. You may be charged more than the amount stated above if you fail to do so.

**Comment by chickenn00dle** [comment]
> yes exactly I am getting this type of URL ?woo-paypal-return=true&token is there any solution to resolve this? please

As I mentioned, this is expected behavior and this is not something that can be changed.

> Since your browser does not support JavaScript, or it is disabled, please ensure you click the Update Totals button before placing your order. You may be charged more than the amount stated above if you fail to do so. 

It might be a good idea to [reach out to our support team](https://woocommerce.com/my-account/create-a-ticket/?select=1597922&_ga=2.196918572.185884012.1630649692-1324334147.1618278716&_gac=1.191965912.1629347439.CjwKCAjw3_KIBhA2EiwAaAAlimf7kjyue_do1r0-7B1v5O_eMZKsPN83GZHri0kr170UMPYFSHuKQRoClk0QAvD_BwE) who can help with store-specific issues.

Issues in this repository are for reporting more general issues specific to the plugin.

**Comment by rajeshraval786** [comment]
Ok, Thanks for giving your valuable time.
